packages/core/src/base/EngineObject.ts

```diff
@@ -15,14 +15,35 @@ export abstract class EngineObject {
   @ignoreClone
   protected _engine: Engine;
 
+  protected _destroyed: boolean = false;
+
   /**
    * Get the engine which the object belongs.
    */
   get engine(): Engine {
     return this._engine;
   }
 
+  /**
+   * Whether it has been destroyed.
+   */
+  get destroyed(): boolean {
+    return this._destroyed;
+  }
+
   constructor(engine: Engine) {
     this._engine = engine;
   }
+
+  /**
+   * Destroy self.
+   */
+  destroy(): void {
+    if (this._destroyed) return;
+
+    const resourceManager = this._engine.resourceManager;
+    resourceManager?._deleteAsset(this);
```

Comment:
Need not to create a variable used once:
```typescript
this._engine.resourceManager?._deleteAsset(this);
```

Reply:
done